File: Programação orientada a objetos/POO 241 - Trab 1 - Pedro Quedevez/Publicacao.java
import java.util.ArrayList;

public class Publicacao {

    private int ano;
    private String siglaVeiculo;
    private String titulo;
    private ArrayList<Long> listaAutores;
    private int numero;
    private int volume;
    private String localConferencia;
    private int paginaInicial;
    private int paginaFinal;
    private String quali;

    public Publicacao(String csvLine) throws excecao {
        String[] campos = csvLine.split(";");
        for (int i = 0; i < campos.length; i++) {
            if (campos[i].equals("")) {
                campos[i] = "-1"; // ta servindo pra algo ainda?
            }
        }
        try {
            if (campos.length == 9) {
                this.ano = Integer.parseInt(campos[0].trim());
                if (!Veiculo.siglasCadastradas.contains(campos[1].trim())) {
                    throw new excecao(
                            "Sigla de veículo não definida usada na publicação " + "\"" + campos[2] + "\"" + ": "
                                    + campos[1]
                                    + ".");
                }
                this.siglaVeiculo = campos[1].trim();
                this.titulo = campos[2].trim();
                this.listaAutores = new ArrayList<Long>();
                String[] autores = campos[3].split(",");
                for (String autor : autores) {
                    Long autorLong = Long.parseLong(autor.trim()); // algumas strings tem espaços no final;
                    if (!Docente.codigosCadastrados.contains(autorLong)) {
                        throw new excecao(
                                "Código de docente não definido usado na publicação " + "\"" + campos[2] + "\"" + ": "
                                        + autorLong
                                        + ".");
                    }
                    this.listaAutores.add(autorLong);
                }
                this.numero = Integer.parseInt(campos[4].trim());
                this.volume = Integer.parseInt(campos[5].trim());
                this.localConferencia = campos[6].trim();
                this.paginaInicial = Integer.parseInt(campos[7].trim());
                this.paginaFinal = Integer.parseInt(campos[8].trim());
                this.quali = "";
            }
        } catch (java.lang.NumberFormatException e) {
            throw new excecao("Erro de formatacao");
        }
    }

    @Override
    public String toString() {
        String autores = "";
        for (Long autor : listaAutores) {
            autores += autor + ",";
        }
        return ano + ";" + siglaVeiculo + ";" + titulo + ";" + autores + ";" + numero + ";" + volume + ";"
                + localConferencia + ";" + paginaInicial + ";" + paginaFinal;
    }

    public int getAno() {
        return ano;
    }

    public void setAno(int ano) {
        this.ano = ano;
    }

    public String getSiglaVeiculo() {
        return siglaVeiculo;
    }

    public void setSiglaVeiculo(String siglaVeiculo) {
        this.siglaVeiculo = siglaVeiculo;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public ArrayList<Long> getListaAutores() {
        return listaAutores;
    }

    public void setListaAutores(ArrayList<Long> listaAutores) {
        this.listaAutores = listaAutores;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public int getVolume() {
        return volume;
    }

    public void setVolume(int volume) {
        this.volume = volume;
    }

    public String getLocalConferencia() {
        return localConferencia;
    }

    public void setLocalConferencia(String localConferencia) {
        this.localConferencia = localConferencia;
    }

    public int getPaginaInicial() {
        return paginaInicial;
    }

    public void setPaginaInicial(int paginaInicial) {
        this.paginaInicial = paginaInicial;
    }

    public int getPaginaFinal() {
        return paginaFinal;
    }

    public void setPaginaFinal(int paginaFinal) {
        this.paginaFinal = paginaFinal;
    }

    public String getQuali() {
        return quali;
    }

    public void setQuali(String quali) {
        this.quali = quali;
    }

}

File: Programação orientada a objetos/POO-241 - Trab 2 - Pedro_Quedevez/Publicacao.cpp

#include <iostream>
#include <string>
#include <vector>
#include "StringUtils.hpp"
#include "Tokenizer.hpp"
#include "Publicacao.hpp"
#include "Veiculo.hpp"
#include "Excecao.hpp"
#include "Docente.hpp"

using namespace cpp_util;
using namespace std;

Publicacao::Publicacao(string csvLine)
{

    Tokenizer tokens(csvLine, ';');
    vector<string> campos = tokens.remaining();

    try
    {

        for (int i = 0; i < campos.size(); i++)
        {
            campos[i] = trim(campos[i]);
        }

        this->setAno(stoi(campos[0]));
        if (siglasUtilizadas.find(trim(campos[1])) == siglasUtilizadas.end())
        {
            throw Excecao("Sigla de veículo não definida usada na publicação \"" + campos[2] + "\": " + campos[1] + ".");
        }
        this->setSiglaVeiculo(campos[1]);
        this->setTitulo(campos[2]);
        this->setAutores(campos[3]);

        Tokenizer autores(campos[3], ',');
        vector<string> autoresSeparados = autores.remaining();
        for (int i = 0; i < autoresSeparados.size(); i++)
        {
            if (codigosUtilizados.find(trim(autoresSeparados[i])) == codigosUtilizados.end())
            {
                throw Excecao("Código de docente não definido usado na publicação \"" + campos[2] + "\": " + autoresSeparados[i] + ".");
            }
        }

        this->setNumero(stoi(campos[4]));
        if (campos[5] == "")
        {
            campos[5] = "0";
        }
        this->setVolume(stoi(campos[5]));
        this->setLocalConferecia(campos[6]);
        this->setPaginaInicial(stoi(campos[7]));
        this->setPaginaFinal(stoi(campos[8]));
        this->setQuali("");
    }
    catch (Excecao e)
    {
        throw e;
    }
}

int Publicacao::getAno()
{
    return this->ano;
}
string Publicacao::getSiglaVeiculo()
{
    return this->siglaVeiculo;
}
string Publicacao::getTitulo()
{
    return this->titulo;
}
string Publicacao::getAutores()
{
    return this->autores;
}
int Publicacao::getNumero()
{
    return this->numero;
}
int Publicacao::getVolume()
{
    return this->volume;
}
string Publicacao::getLocalConferecia()
{
    return this->localConferecia;
}
int Publicacao::getPaginaInicial()
{
    return this->paginaInicial;
}
int Publicacao::getPaginaFinal()
{
    return this->paginaFinal;
}
string Publicacao::getQuali()
{
    return this->quali;
}

Veiculo Publicacao::getVeiculo()
{
    return this->V;
}

void Publicacao::setVeiculo(Veiculo v)
{

    this->V = v;
}

void Publicacao::setAno(int i)
{
    this->ano = i;
}
void Publicacao::setSiglaVeiculo(string s)
{
    this->siglaVeiculo = s;
}
void Publicacao::setTitulo(string s)
{
    this->titulo = s;
}
void Publicacao::setAutores(string v)
{
    this->autores = v;
}
void Publicacao::setNumero(int i)
{
    this->numero = i;
}
void Publicacao::setVolume(int i)
{
    this->volume = i;
}
void Publicacao::setLocalConferecia(string s)
{
    this->localConferecia = s;
}
void Publicacao::setPaginaInicial(int i)
{
    this->paginaInicial = i;
}
void Publicacao::setPaginaFinal(int i)
{
    this->paginaFinal = i;
}
void Publicacao::setQuali(string s)
{
    this->quali = s;
}

ostream &operator<<(ostream &c, Publicacao &d)
{
    c << d.getAno() << " " << d.getSiglaVeiculo() << " " << d.getTitulo() << " " << d.getAutores() << " " << d.getNumero() << " " << d.getVolume() << " " << d.getLocalConferecia() << " " << d.getPaginaInicial() << " " << d.getPaginaFinal() << " " << d.getQuali();
    return c;
}